Add expand/compress image button on image view box

mashirozx opened the pull request:
Add feature for #15018 

A live demo here: https://hello.2heng.xin/@mashiro/105123931762135694

## Feature
- Add a expand/compress image button on image view box (besides the close button)
- While the image being zoomed, view box can be scrolled by mouse grabbing

## Motivation
Consider the condition that there is a very long image, it's hard to scale and read on desktop:

![demo1](https://user-images.githubusercontent.com/16148054/97712105-f46e1a00-1af8-11eb-9cee-c36647558a76.png)

In this PR, we add a button that can scale image on click, to fit the width/height (depend on image's width/height ratio) of the screen so image can be read easily though scroll:

![demo2](https://user-images.githubusercontent.com/16148054/97711859-94777380-1af8-11eb-9d60-708166b263c9.png)

mashirozx committed:
add zoom image button

mashirozx committed:
enhance zoom algorithm & add translation

mashirozx committed:
code structure

mashirozx committed:
code structure

mashirozx committed:
code structure

mashirozx committed:
enhance grab performance

mashirozx committed:
rm useless state

Gargron commented:
I have checked out the demo and noticed an issue. When you expand the image and try to pan it by dragging, the browser tries to drag the image file. Then when you release the mouse button, the image starts following the cursor until you click.

Gargron commented:
Intuitively, I would rely on the pan-by-dragging behaviour instead of scrollbars or mouse wheel. It takes a lot of effort to scroll somewhere using the mouse wheel on a long image, and dragging the scrollbar thumb is buggy because the scrollbar overlays with other modal elements. Plus, the scrollbars aren't very visually appealing. So I would suggest hiding scrollbars and fixing the pan-by-dragging functionality.

mashirozx committed:
fix behavior on Firefox & scroll lock & horizontal scroll with mousewheel

mashirozx commented:
@Gargron 
- I fixed the drag behavior on Firefox
- hidden the scrollbars
- added a feature of scroll long width image horizontal with mouse wheel
- locked the min `scrollLeft` and `scrollTop` to hide the ugly blank area

If it's convenient for you, please have a try. My demo site need some time to build Docker image, so I suggest to run PR code with a dev server 😄

If it's functionality OK, I'll enhance the code structure.

mashirozx committed:
remove scroll lock on MouseWheelEvent

mashirozx committed:
code structure

mashirozx committed:
enhance algorithm and code structure

mashirozx committed:
rm Gemfile.lock from tree

mashirozx committed:
codeclimate

mashirozx committed:
fix a stupid mistake

Krinkle commented:
In the context of translation for Crowdin, I'd like to understand what "compress" means in this context. I assume it is not related to image file compression, but something visual in the lightbox. Does it mean close? Or minimize? Or shrink? Perhaps one of those might also work better in English. Thank you!

Gargron commented:
> For now, based on the demo, I have loosely translated these in Dutch as "Enlarge image" and "Fit image".

@Krinkle This works! Yes, it's about how the picture is displayed in the browser window.

mashirozx commented:
> For now, based on the demo, I have loosely translated these in Dutch as "Enlarge image" and "Fit image".

@Krinkle I can't agree more on this! 😀
